app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -45,17 +53,50 @@ public static String generateRandomId() {
     return sb.toString();
   }
 
-  public static void addDocument(String key, Map document, String collection)
+  public static void addDocument(String key, Map document, String collection) {
+    new TaskToFutureAdapter<>(getCollectionReference(collection).document(key).set(document));
+  }
+
+  public static void removeDocument(String key, String collection) {
+    Task<Void> deleteTask = getCollectionReference(collection).document(key).delete();
+    new TaskToFutureAdapter<>(deleteTask);
+  }
+
+  public static void updateDocument(String key, Map<String, Object> updates, String collection) {
+    Task<Void> updateTask = getCollectionReference(collection).document(key).update(updates);
+    new TaskToFutureAdapter<>(updateTask);
+  }
+
+  public static CompletableFuture<Map> getDocument(String key, String collection)
       throws RuntimeException {
-    DatabaseWrapper.getInstance()
-        .firestore
-        .collection(collection)
-        .document(key)
-        .set(document)
-        .addOnSuccessListener(aVoid -> {})
-        .addOnFailureListener(
-            e -> {
-              throw new RuntimeException(e);
-            });
+    Task<DocumentSnapshot> getTask = getCollectionReference(collection).document(key).get();
+    CompletableFuture<DocumentSnapshot> future = new TaskToFutureAdapter<>(getTask);
+
+    return future.thenApply(
+        documentSnapshot -> {
+          if (documentSnapshot.exists()) {
+            return documentSnapshot.getData();
```

Comment:
Here you get a Map back. I guess you will be using your `CollectionWrapper` to convert it back to your data models? Otherwise you could pass a type parameter T to supply a constructor from map to T.

Reply:
Changed it to use type parameters relevant to our code.